File: src/main/java/me/jollyperson/featurebot/handlers/Command.java
package me.jollyperson.featurebot.handlers;

import java.util.List;

public interface Command {
    void handle(CommandContext ctx);

    String getName();

    String getHelp();

    default List<String> getAliases() {
        return List.of(); // use Arrays.asList if you are on java 8
    }
}

File: src/main/kotlin/me/jollyperson/featurebot/handlers/Command.kt
package me.jollyperson.featurebot.handlers

import me.jollyperson.featurebot.objects.PermissionFlag
import java.util.*

interface Command {
    fun handle(ctx: CommandContext)
    val name: String
    val help: String?
    val aliases: List<String?>
        get() = Arrays.asList()

    val permissions: List<PermissionFlag>
}